bdk_core/src/chain_graph.rs

```diff
@@ -32,25 +32,38 @@ impl<I> Default for ChainGraph<I> {
     }
 }
 
+impl<I> ChainGraph<I> {
+    pub fn chain(&self) -> &sparse_chain::SparseChain<I> {
+        &self.chain
+    }
+
+    pub fn graph(&self) -> &TxGraph {
+        &self.graph
+    }
+}
+
 impl<I: ChainIndex> ChainGraph<I> {
     pub fn insert_tx(
         &mut self,
         tx: Transaction,
-        index: I,
-    ) -> Result<bool, sparse_chain::InsertTxErr> {
-        let changed = self.chain.insert_tx(tx.txid(), index)?;
-        self.graph.insert_tx(tx);
-        Ok(changed)
+        index: Option<I>,
+    ) -> Result<InsertOk, sparse_chain::InsertTxErr> {
+        let chain_changed = match index {
+            Some(index) => self.chain.insert_tx(tx.txid(), index)?,
+            None => false,
+        };
+        let graph_changed = self.graph.insert_tx(tx);
+
+        Ok(InsertOk {
+            chain: chain_changed,
+            graph: graph_changed,
+        })
```

Comment:
I can see why you are returning this but I don't think we need it. You can figure out whether a tx is changing height from the outside in the odd case you really need to know this. Changes where you need to record them are done with changesets.

Reply:
I think I will leave this as is for now